src/protocol/sort/bit_permutation.rs

```diff
@@ -50,9 +50,13 @@ where
             Some((x, sum.clone()))
         });
 
-    let async_multiply = zip(repeat(ctx), mult_input)
-        .enumerate()
-        .map(|(i, (ctx, (x, sum)))| async move { ctx.multiply(RecordId::from(i), &x, &sum).await });
+    let async_multiply =
+        zip(repeat(ctx), mult_input)
+            .enumerate()
+            .map(|(i, (ctx, (x, sum)))| async move {
+                let record_id = RecordId::from(i);
+                ctx.bind(record_id).multiply(record_id, &x, &sum).await
```

Comment:
makes sense

Reply:
You don't need to bind the record_id here, do you?